db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/SessionPropertiesImpl.java

```diff
@@ -104,11 +111,59 @@ public void setProperty(PROPERTYNAME property, Object value) {
                 boolean useNativeSpark = Boolean.parseBoolean(valString);
                 properties[USE_NATIVE_SPARK.getId()] = useNativeSpark;
                 break;
+            case CURRENTFUNCTIONPATH:
+                properties[CURRENTFUNCTIONPATH.getId()] = parseCurrentFunctionPath(valString);
+                break;
             default:
                 assert false;
         }
     }
 
+    /**
+     * parses the value of CURRENTFUNCTIONPATH according to:
+     * https://www.ibm.com/support/producthub/db2/docs/content/SSEPGG_11.5.0/com.ibm.db2.luw.apdv.cli.doc/doc/r0008777.html
+     * @param value the user-input value
+     * @return list of schemas for function lookup.
+     */
+    private String[] parseCurrentFunctionPath(String value) {
+        if(value == null || value.isEmpty()) {
+            return new String[] {"SYSIBM", "SYSFUN"};
+        } else {
+            // source: https://stackabuse.com/regex-splitting-by-character-unless-in-quotes/
+            List<String> candidateSchemas = new ArrayList<>(Arrays.asList(value.split(",(?=([^\"]*\"[^\"]*\")*[^\"]*$)")));
+            for(int i = 0; i < candidateSchemas.size(); ++i) {
+                candidateSchemas.set(i , normalizeAndUnquoteIdentifier(candidateSchemas.get(i)));
+            }
+            boolean sysibmSeen = false;
+            boolean sysfunSeen = false;
+            for(int i = 0; i < candidateSchemas.size(); ++i) {
```

Comment:
not sure if it's critical, but if for example SYSIBM will not on the first position, it will be added again to the first place.
input = [AAA, SYSIBM, BBB]
output = [SYSIBM, SYSFUN, AAA, SYSIBM, BBB]
there will be 2 items SYSIBM in the list, Maybe better to use indexOf method and move/add to the first place.

Reply:
Thanks Igor for catching this, it is not clear to me from [DB2 documentation](https://www.ibm.com/support/producthub/db2/docs/content/SSEPGG_11.5.0/com.ibm.db2.luw.apdv.cli.doc/doc/r0008777.html) what's exactly the expected behavior in this case but I think it might be ok to leave it like this because if we are searching for function e.g. `cos` then we have two cases:

1. it exists in `SYSIBM` therefore we call it first regardless if `SYSIBM` was manually added by the user in some order.
2. it doesn't exist in `SYSIBM`, then we skip the first `SYSIBM` and any other occurance of it potentially specified by the user.

Therefore in both cases, it doesn't matter much if the user specified it that's why I just follow what DB2 says:

> The default value is "SYSIBM","SYSFUN",X where X is the value of the USER special register delimited by double quotation marks

What do you think?

Reply:
it's ok for me to keep so